### ISSUE TITLE
list the inherited attributes

### ISSUE THREAD

**Issue by roberbnd** [liuchengxu/vista.vim#396]
I would like to see when I open a class file a section with the methods or a window (because the list is big) showing the  inherited attributes/methods with PHP I am using `nvim-lua/completion-nvim` then if I write $this-> shows all the values but I would like to see it  with this plugin.

**Comment by liuchengxu** [comment]
I don't follow, can you rephrase that?

**Comment by roberbnd** [comment]
Intellij IDE has these options on 'structure' section
<img width="468" alt="Screen Shot 2021-04-10 at 10 10 56" src="https://user-images.githubusercontent.com/4907027/114274763-9cabd700-99e5-11eb-8d54-c244a5d62cc1.png">
it's possible to see the inherited attributes and filter other this, I want that, of course it's your time, if you don't want it or you don't have time, Could you guide me to implement?

**Comment by liuchengxu** [comment]
First of all, you have to use LSP or tree-sitter(https://github.com/liuchengxu/vista.vim/issues/395, not yet supported) to understand the semantics as ctags is only regexp based and has a big flaw in tree view rendering(#320), it's not reliable to obtain the info of inherited attributes from ctags. You can see the currently supported executives in https://github.com/liuchengxu/vista.vim/tree/master/autoload/vista/executive. Except `ctags`, all the other executives are LSP client for retrieving the LSP info, but only the `coc` executive supports the tree view now.

The basic idea of this plugin is to request the original symbols info from these executives/backends, process this info, and render them in tree view if possible. The tree renderer is in https://github.com/liuchengxu/vista.vim/tree/master/autoload/vista/renderer/hir. You can first take a look how vista.vim records the symbols and then you can filter out the symbols you want.
